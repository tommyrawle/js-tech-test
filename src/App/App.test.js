import React from 'react';
import { shallow } from 'enzyme';
import '@babel/polyfill';

let wrapper;
const mockHistoryPush = jest.fn();
describe('App component', () => {
  describe('Happy route', () => {
    beforeEach(() => {
      jest.resetModules();
      jest.mock('react-router-dom', () => ({
        ...jest.requireActual('react-router-dom'),
        useHistory: () => ({
          push: mockHistoryPush
        }),
        useLocation: () => ({
          pathname: '/'
        })
      }));
    });
    it('should render Spinner if app hasnt initialised yet', () => {
      const mockProps = {
        setError: jest.fn(),
        error: {},
        initialiseApp: jest.fn(),
        isAppInitialised: false
      };
      const { App } = require('./App.js');
      wrapper = shallow(<App {...mockProps} />);
      expect(wrapper).toMatchSnapshot();
      expect(wrapper.find('Spinner').exists()).toBeTruthy();
    });
    it('should render main app once app has initialised', () => {
      const mockProps = {
        setError: jest.fn(),
        error: {},
        initialiseApp: jest.fn(),
        isAppInitialised: true
      };
      const { App } = require('./App.js');
      wrapper = shallow(<App {...mockProps} />);
      expect(wrapper).toMatchSnapshot();
      expect(wrapper.find('Spinner').exists()).toBeFalsy();
    });
  });

  describe('Error route', () => {
    beforeEach(() => {
      jest.resetModules();
      jest.mock('react-router-dom', () => ({
        ...jest.requireActual('react-router-dom'),
        useHistory: () => ({
          push: mockHistoryPush
        }),
        useLocation: () => ({
          pathname: '/error'
        })
      }));
      const { App } = require('./App.js');
      const mockProps = {
        setError: jest.fn(),
        error: { code: 500, message: 'something bad happened' }
      };
      wrapper = shallow(<App {...mockProps} />);
    });
    it('should render Error route if pathname is /error', async () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
