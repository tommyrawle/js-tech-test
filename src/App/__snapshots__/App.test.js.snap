// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component Error route should render Error route if pathname is /error 1`] = `
<div>
  <Header
    title="Live Events"
  />
  <Route
    path="/error"
  >
    <Connect(Error) />
  </Route>
</div>
`;

exports[`App component Happy route should render Spinner if app hasnt initialised yet 1`] = `<Spinner />`;

exports[`App component Happy route should render main app once app has initialised 1`] = `
<div>
  <Header
    title="Live Events"
  />
  <Switch>
    <Route
      exact={true}
      path="/"
    >
      <Connect(EventList) />
    </Route>
    <Route
      path="/event/:eventId"
    >
      <Connect(Event) />
    </Route>
    <Route
      path="/error"
    >
      <Connect(Error) />
    </Route>
    <Route
      path="*"
    >
      <Connect(Error) />
    </Route>
  </Switch>
</div>
`;
